server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -99,14 +99,14 @@ protected void masterOperation(Task task, final RolloverRequest rolloverRequest,
                                    final ActionListener<RolloverResponse> listener) throws Exception {
         MetadataRolloverService.RolloverResult preResult =
             rolloverService.rolloverClusterState(state,
-                rolloverRequest.getAlias(), rolloverRequest.getNewIndexName(), rolloverRequest.getCreateIndexRequest(),
+                rolloverRequest.getRolloverTarget(), rolloverRequest.getNewIndexName(), rolloverRequest.getCreateIndexRequest(),
```

Comment:
The `TransportRolloverAction#checkBlock()` method should be changed.

The following

```
IndicesOptions indicesOptions = IndicesOptions.fromOptions(true, true,
            request.indicesOptions().expandWildcardsOpen(), request.indicesOptions().expandWildcardsClosed());
```

should be changed into:

```
IndicesOptions indicesOptions = IndicesOptions.fromOptions(true, true,
            request.indicesOptions().expandWildcardsOpen(), request.indicesOptions().expandWildcardsClosed(), 
            request.indicesOptions().expandWildcardsHidden(), request.indicesOptions().allowAliasesToMultipleIndices(), 
            request.indicesOptions().forbidClosedIndices(), request.indicesOptions().ignoreAliases(), 
            request.indicesOptions().ignoreThrottled(), request.indicesOptions().includeDataStreams());
```

This why the modified test in `DataStreamIT` failed.